frontend/lib/src/components/widgets/CustomComponent/ComponentInstance.test.tsx

```diff
@@ -109,185 +89,86 @@ class MockComponent {
+        width={100}
+        disabled={false}
+        theme={mockTheme.emotion}
+        widgetMgr={
+          new WidgetStateManager({
+            sendRerunBackMsg: jest.fn(),
+            formsDataChanged: jest.fn(),
+          })
+        }
+      />
     )
+    unmount()
+    expect(deregisterListener).toHaveBeenCalledTimes(1)
   })
 
   it("renders its iframe correctly", () => {
-    // This is not an exhaustive check of rendering props - instead, it's
-    // the props whose values are functionally important.
-    const mc = new MockComponent()
-    const iframe = mc.wrapper.childAt(0)
-    expect(iframe.type()).toEqual("iframe")
-    expect(iframe.prop("src")).toContain(MOCK_COMPONENT_URL)
-    expect(iframe.prop("allow")).toEqual(DEFAULT_IFRAME_FEATURE_POLICY)
-    expect(iframe.prop("sandbox")).toEqual(DEFAULT_IFRAME_SANDBOX_POLICY)
+    const componentRegistry = getComponetRegistry()
+    render(
+      <ComponentInstance
+        element={createElementProp()}
+        registry={componentRegistry}
+        width={100}
+        disabled={false}
+        theme={mockTheme.emotion}
+        widgetMgr={
+          new WidgetStateManager({
+            sendRerunBackMsg: jest.fn(),
+            formsDataChanged: jest.fn(),
+          })
+        }
+      />
+    )
+    const iframe = screen.getByTitle(MOCK_COMPONENT_NAME)
+    expect(iframe).toHaveAttribute(
+      "src",
+      "http://a.mock.url?streamlitUrl=http%3A%2F%2Flocalhost%2F"
+    )
+    expect(iframe).toHaveAttribute("allow", DEFAULT_IFRAME_FEATURE_POLICY)
+    expect(iframe).toHaveAttribute("sandbox", DEFAULT_IFRAME_SANDBOX_POLICY)
   })
 
-  it("sends JSON args to iframe", () => {
-    const mc = new MockComponent()
-
-    // We should receive an initial RENDER message with no arguments
-    mc.sendBackMsg(ComponentMessageType.COMPONENT_READY, { apiVersion: 1 })
-    expect(mc.receiveForwardMsg).toHaveBeenLastCalledWith(
-      renderMsg({}, []),
-      "*"
-    )
+  // TODO: implement test to check sends dataframe args to iframe
 
-    // Simulate passing JSON arguments to the component.
+  it("renders with json arguments", () => {
     const jsonArgs = { foo: "string", bar: 5 }
     const element = createElementProp(jsonArgs, [])
-    mc.wrapper.setProps({ element })
-
-    expect(mc.instance.state.componentError).toBeUndefined()
-
-    // We should get those JSON arguments in our receiveForwardMsg callback.
-    expect(mc.receiveForwardMsg).toHaveBeenLastCalledWith(
-      renderMsg(jsonArgs, []),
-      "*"
+    render(
+      <ComponentInstance
+        element={element}
+        registry={getComponetRegistry()}
+        width={100}
+        disabled={false}
+        theme={mockTheme.emotion}
+        widgetMgr={
+          new WidgetStateManager({
+            sendRerunBackMsg: jest.fn(),
+            formsDataChanged: jest.fn(),
+          })
+        }
+      />
     )
+    expect(screen.getByTitle(MOCK_COMPONENT_NAME)).toBeVisible()
```

Comment:
Might not describe this test renders the component well without testing if the contents of the iframe contain the arguments.